### ISSUE TITLE
NameError: global name 'numb' is not defined

### ISSUE THREAD

**Issue by rofl0r** [Gallopsled/pwntools#974]
```
Traceback (most recent call last):
  File "foo.py", line 907, in <module>
    inp = sys.stdin.readline().strip().decode('utf-8')
  File "/lib/python2.7/site-packages/pwnlib/term/readline.py", line 412, in readline
    return readline(size)
  File "/lib/python2.7/site-packages/pwnlib/term/readline.py", line 376, in readline
    keymap.handle_input()
  File "/lib/python2.7/site-packages/pwnlib/term/keymap.py", line 20, in handle_input
    self.send(key.get())
  File "/lib/python2.7/site-packages/pwnlib/term/key.py", line 164, in get
    k = _peek()
  File "/lib/python2.7/site-packages/pwnlib/term/key.py", line 159, in _peek
    return _peek_ti() or _peek_csi() or _peek_simple()
  File "/lib/python2.7/site-packages/pwnlib/term/key.py", line 393, in _peek_csi
    return _peekkey_ss3(2)
  File "/lib/python2.7/site-packages/pwnlib/term/key.py", line 372, in _peekkey_ss3
    _cbuf = _cbuf[numb:] # XXX: numb is not defined
NameError: global name 'numb' is not defined

```
this happens when i press + and enter on the numeric block without having numlock on.
also kinda weird even that this is handled by pwnlib, since i call a function from sys.

**Comment by rofl0r** [comment]
from that point on stdin is dead, any try to access it causes this exception, even if caught.

**Comment by rofl0r** [comment]
this happens quite a lot since sometimes when i enter a command into my program i instinctively hit enter on the numpad. from that point on my application becomes unusable and i have to kill it. no workaround found so far.

**Comment by zachriggle** [comment]
I can't reproduce this.  Please provide steps that I can reproduce in the docker image.

```
>>> from pwn import *
>>> inp = sys.stdin.readline().strip().decode('utf-8')
asdfasdfasdf
>>> inp
u'asdfasdfasdf'
```

To run in Docker:

```
$ docker pull pwntools/pwntools:stable
$ docker run -it pwntools/pwntools:stable
pwntools@e5bc67ee7449:~$ python
Python 2.7.12 (default, Nov 19 2016, 06:48:10)
[GCC 5.4.0 20160609] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> from pwn import *
>>> sys.stdin.readline().strip().decode("utf-8")
Hello, world!
u'Hello, world!'
```

**Comment by rofl0r** [comment]
this causes the bug for me:
test.py
```py
from pwn import *
import sys

inp = sys.stdin.readline().strip().decode('utf-8')
```
`python test.py`
hit + on numeric block with numlock off

i'm sorry but i cannot use docker, since my [custom linux distribution based on musl libc](https://github.com/sabotage-linux/sabotage) doesn't yet bundle a go compiler. but even if it's not reproducible on a mainstream distro, the bug is real as can be seen from the fat XXX in the comment...

**Comment by rofl0r** [comment]
note that i cannot reproduce the issue either in an interactive python session - it must be run from a script

**Comment by rofl0r** [comment]
a member of my team just tried it on debian, and he was able to reproduce the issue running the above script. *sigh of relief*

**Comment by zachriggle** [comment]
I still cannot reproduce this on either macOS or Ubuntu 16.04.

```
$ pwn update
[*] You have the latest version of Pwntools (3.5.1)
$ cat x.py
from pwn import *
import sys

inp = sys.stdin.readline().strip().decode('utf-8')
print(repr(inp))
$ python x.py
asdfasdfasdf
u'asdfasdfasdf'
```

**Comment by zachriggle** [comment]
I don't have any computers with a full keyboard, so no number pad to hit the `+` key on, or to turn NumLock off.

**Comment by rofl0r** [comment]
it should be sufficient to just hit "+" on numeric block. my friends debian version is 
```
$ cat /etc/debian_version
8.5
```

**Comment by rofl0r** [comment]
damn, in that case i guess you won't be able to reproduce it! no usb keyboard lying around somewhere ?

**Comment by zachriggle** [comment]
Nope! :-\

**Comment by zachriggle** [comment]
In any case, this is in the `term` code which I don't touch.

**Comment by zachriggle** [comment]
It looks like that code (minus the XXX warning, which was added later) comes from ab3d128a776ec29faf281f33644fecdb32e79d3e

```
commit ab3d128a776ec29faf281f33644fecdb32e79d3e
Author: Morten Brøns-Pedersen <f@ntast.dk>
Date:   Wed Apr 30 14:30:21 2014 +0200

    started working on pwnlib v. 2.0
```

**Comment by rofl0r** [comment]
hey @br0ns , do you have an idea why this code is interfering with my invocation of sys code ? this looks like quite a nasty hook to me.

**Comment by zachriggle** [comment]
You can disable term mode by setting PWNLIB_NOTERM in your environment, or
invoking your script with "python foo.py NOTERM". This will disable all
animations and fancy terminal things, and also resolve your issue.

**Comment by rofl0r** [comment]
thanks! when looking at the issues/PRs here, i have the feeling that assigning @br0ns to a task is about the same as closing it with EWONTFIX.. :)

**Comment by rofl0r** [comment]
@zachriggle: is there a way to set this from code, rather than changing the environment or script args ? i'd rather do context.term_mode = False in my code and have the issue fixed once and for all, than having to start my program from a shell wrapper so i dont forget to set the env var.

**Comment by zachriggle** [comment]
Unfortunately the term initialization opt-out *must* pre-empt "from pwn
import *"

**Comment by br0ns** [comment]
I've made a PR with an attempt at a fix: https://github.com/Gallopsled/pwntools/pull/978

The code is based on [libtermkey](https://github.com/neovim/libtermkey).

**Comment by rofl0r** [comment]
cool! i confirm that the issue is now fixed. if numlock is off, and "+" is being pressed "+k" is shown in the terminal instead of crashing. thank you!

**Comment by zachriggle** [comment]
Merged to `stable` in #979; this fix will be in `3.6.1`
